Add more tests for LayerStore

chrismayer opened the pull request:
Adding more tests for the ``GeoExt.data.store.Layers`` class.

chrismayer commented:
Anyone willing to do a review? Thanks in advance!

marcjansen commented on test/spec/GeoExt/data/store/Layers.test.js:
```diff
@@ -6,5 +6,141 @@ describe('GeoExt.data.store.Layers', function() {
+
+    describe('function "unbindMap"', function() {
+        var store;
+        beforeEach(function(done) {
```
Do we need the `done` and further down below? If not, please remove.

marcjansen commented on test/spec/GeoExt/data/store/Layers.test.js:
```diff
@@ -6,5 +6,141 @@ describe('GeoExt.data.store.Layers', function() {
+        });
+
+        it('removes the map', function() {
+            expect(store.map).to.be.null;
```
Did you mean to write `to.be(null)`?

And how would that pass?  Don't you need to call `unbindMap` (or do sth. that then calls `unbindMap`)?

marcjansen commented on test/spec/GeoExt/data/store/Layers.test.js:
```diff
@@ -6,5 +6,141 @@ describe('GeoExt.data.store.Layers', function() {
+    describe('function "getByLayer"', function() {
+        var store;
+        var layer;
+        beforeEach(function(done) {
```
Please check if `done` is needed.

marcjansen commented on test/spec/GeoExt/data/store/Layers.test.js:
```diff
@@ -6,5 +6,141 @@ describe('GeoExt.data.store.Layers', function() {
+        var map;
+        var layer1;
+        var layer2;
+        beforeEach(function(done) {
```
Please check if `done` is needed.

marcjansen commented on test/spec/GeoExt/data/store/Layers.test.js:
```diff
@@ -6,5 +6,141 @@ describe('GeoExt.data.store.Layers', function() {
+
+        describe('custom "bind" event', function() {
+            var i = 0;
+            beforeEach(function(done) {
```
Please check if `done` is needed.

marcjansen commented:
I have added a couple of tiny comments and one question for the `unbindMap` tests.

Once these are adressed and the checks are green, this is good to go. Thanks @chrismayer

chrismayer commented:
Thanks @marcjansen for your review. I addressed all your comments and adapted the original code. Would you have another look please? Would be cool! TIA!

marcjansen commented:
This looks good to me. Thanks and please merge, @chrismayer
